### PR TITLE
fixes #19925 - bulk content hosts repo sets

### PR DESCRIPTION
**Pull request by thomasmckay**
PR relies on https://github.com/Katello/katello/pull/6809 (included as separate commit here to be removed before merge)

### CODE REVIEW

File: app/controllers/katello/api/v2/hosts_bulk_actions_controller.rb [hunk 1]
```diff
@@ -176,7 +176,7 @@ def auto_attach
       respond_for_async :resource => task
     end
 
-    api :PUT, "/hosts/bulk/subscriptions/content_overrides", N_("Set content overrides to one or more hosts")
+    api :PUT, "/hosts/bulk/content_overrides", N_("Set content overrides to one or more hosts")
```
Comment (jlsherrill) [inline app/controllers/katello/api/v2/hosts_bulk_actions_controller.rb]: The documented route was wrong, correct?

Comment (thomasmckay) [inline app/controllers/katello/api/v2/hosts_bulk_actions_controller.rb]: Correct, the route didn't exist as documented here.

File: app/controllers/katello/api/v2/repository_sets_controller.rb [hunk 2]
```diff
@@ -83,18 +88,32 @@ def find_product_content
...
 
     def substitutions
       params.slice(:basearch, :releasever)
     end
+
+    def available_repository_sets
+      content = Katello::Product.all_products(@organization.id).flat_map do |product|
+        product.available_content
+      end
+      content.uniq
```
Comment (jlsherrill) [inline app/controllers/katello/api/v2/repository_sets_controller.rb]: should we sort by name here?

File: app/models/katello/product.rb [hunk 3]
```diff
@@ -258,5 +258,9 @@ def self.unused_product_id
         id
       end
     end
+
+    def self.all_products(organization_id)
+      ::Katello::Product.joins(:subscriptions => :pools).where(:organization_id => organization_id).enabled.uniq
+    end
```
Comment (jlsherrill) [inline app/models/katello/product.rb]: The name of this method is quite confusing I think.  

Why not just call organization.products.enabled ?  Do you need to join to subscriptions and pools?

**Comment by jlsherrill** [comment]
On the repo sets modal, when i select a repo set, and click 'overide to enabled' or 'disabled', the modal shows no indication that its doing anything, but does not close until the request comes back.  I had to look at the chrome inspector to see what was going on.  Could add some sort of indicator?

File: app/models/katello/product.rb (hunk 3 above)
Comment (thomasmckay) [inline app/models/katello/product.rb]: I do need the join. Suggestions for better name?

**Comment by thomasmckay** [comment]
I noticed that too but it is how the host collections page works as well. I'll have to wait for @waldenraines if we want to fix it in this PR (and should fix on other pages too). I've added an issue for the host collections. I'm fine fixing it in this PR or another.
http://projects.theforeman.org/issues/20009

**Comment by thomasmckay** [comment]
FYI i also noticed this issue http://projects.theforeman.org/issues/19949

File: app/models/katello/product.rb (hunk 3 above)
Comment (jlsherrill) [inline app/models/katello/product.rb]: why do you need the join?

Comment (jlsherrill) [inline app/models/katello/product.rb]: This is the last open concern other than missing the issue # in the commit and subject message

**Comment by jlsherrill** [comment]
ACK pending squash

**Pull request merged**